main.py

```diff
@@ -219,6 +220,24 @@ def ui_access_wrapper(self, *args, **kwargs):
     get_redirect_route(
         r'%s/remove_in_progress_collection' % feconf.LEARNER_DASHBOARD_URL,
         reader.RemoveCollectionFromIncompleteListHandler),
+
+    get_redirect_route(
+        r'%s/add_exploration_to_learner_playlist' %
```

Comment:
It generally doesn't help just to know that something is a "500 error" -- the traceback and error message matter. Did you look at these? What clues did they give?

There is an example of "delete" in the codebase. See the function for deleting explorations in core/controllers/editor.py. Here is a reference on REST that might be useful for background: http://www.restapitutorial.com/lessons/httpmethods.html